### PR TITLE
Universes getter

### PR DESCRIPTION
**Pull request by nuevoalex**
[Clubhouse Story](https://app.clubhouse.io/augur/story/8393)

### CODE REVIEW

File: src/server/getters/get-universes-info.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  if (universe == null || account == null) return callback(new Error("Must provide both universe and account"));
+
+  const query = db.select("parentUniverse").from("universes").where("universe", universe).first();
+
+  query.asCallback((err: Error|null, parentUniverseRow: ParentUniverseRow) => {
+    if (err) return callback(err);
+
+    const query = db.select(["universes.universe", "universes.parentUniverse", "balances.balance", "token_supply.supply", db.raw("count(markets.marketId) as numMarkets")]).from("universes")
+      .where("universes.parentUniverse", universe) // Children
+      .orWhere("universes.parentUniverse", parentUniverseRow.parentUniverse) // Siblings
```
Comment (epheph) [inline src/server/getters/get-universes-info.ts]: If you pass in a universe that does not exit, AN crashes, since this dereferences `undefined`

**Comment by epheph** [comment]
Do we emit a UniverseCreated for genesis?

File: src/server/getters/get-universes-info.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      .orWhere("universes.universe", universe) // Universe
+      .orWhere("universes.universe", parentUniverseRow.parentUniverse) // Parent
+      .leftJoin("token_supply", "token_supply.token", "universes.reputationToken")
+      .leftJoin("balances", function () {
+        this
+          .on("balances.owner", db.raw('"' + account + '"'))
```
Comment (epheph) [inline src/server/getters/get-universes-info.ts]: Should we be concerned about sql injection here?

**Comment by epheph** [comment]
A couple comments, but overall looks great, feel free to merge after those issues are fixed

**Comment by nuevoalex** [comment]
Great point about emitting a log for the genesis universe. Currently we do not. This is an easy change though in the contracts so I'll do that separately.

File: src/server/getters/get-universes-info.ts (hunk 1 above)
Comment (nuevoalex) [reply src/server/getters/get-universes-info.ts]: Will add a test and fix!

File: src/server/getters/get-universes-info.ts (hunk 2 above)
Comment (nuevoalex) [reply src/server/getters/get-universes-info.ts]: Probably. I'll look at the recommended way to do this and avoid sql injection

File: src/server/getters/get-universes-info.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    const query = db.select(["universes.universe", "universes.parentUniverse", "balances.balance", "token_supply.supply", db.raw("count(markets.marketId) as numMarkets")]).from("universes")
+      .where("universes.parentUniverse", universe) // Children
+      .orWhere("universes.parentUniverse", parentUniverseRow.parentUniverse) // Siblings
+      .orWhere("universes.universe", universe) // Universe
+      .orWhere("universes.universe", parentUniverseRow.parentUniverse) // Parent
+      .leftJoin("token_supply", "token_supply.token", "universes.reputationToken")
+      .leftJoin("balances", function () {
+        this
+          .on("balances.owner", db.raw("?", [account]))
```
Comment (epheph) [inline src/server/getters/get-universes-info.ts]: Didn't know you could do that, I actually need to do this on my branch, too!

**Pull request merged**